sound/soc/sof/topology.c

```diff
@@ -2767,9 +2767,9 @@ static int sof_set_dai_config(struct snd_sof_dev *sdev, u32 size,
 			 */
 			config->dai_index = dai->comp_dai.dai_index;
 
-			dai->dai_config = kmemdup(config, size, GFP_KERNEL);
-			if (!dai->dai_config)
-				return -ENOMEM;
+			dai->num_conf = num_conf;
+			dai->def_conf = def_conf;
+			dai->dai_config = config;
```

Comment:
what does this mean?

Are we now saying that each 'config' is now made of <num_conf> configurations?

Also why did we need the kmemdup before and no longer need it?

Reply:
@plbossart yes config is made from num_conf configurations. kmemdup is not needed, because I'm reserving the memory in the calling function... I mean I want to reserve the chunk for all configs, not trying to partially kmemdup here to certain memory locations -> get's messy.